### PR TITLE
feat : added encodeExperiment into jsDOE as per issue #7

### PR DESCRIPTION
**Pull request by aiday-mar**

### CODE REVIEW

**Comment by kjappelbaum** [comment]
Hi!

@lpatiny should `xSampling` not be moved to somewhere in mljs? Alt least in `spectra-processing`?

**Comment by kjappelbaum** [comment]
Perhaps it is even easier if you follow the builder pattern, e.g. https://github.com/tirthajyoti/doepy/tree/master/doepy. 

Then, for some designs, you will only take min and max whereas for others you will consider the full array. 

Only the generalized subset design will work will with the full list of factors as input.

File: src/encodeExperiment.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+export function encodeExperiment(array) {
+  let returnArray = [];
+  let dict = {};
+  let rowNumber = 0;
+
+  for (let i = 0; i < array.length; i++) {
+    let returnRow;
+
+    if ('min' in array[i]) {
```
Comment (lpatiny) [inline src/encodeExperiment.js:11]: If we now the property name we would rather use

array[i].min!==undefined

File: src/encodeExperiment.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      let label = array[i]['label'];
+      let min = array[i]['min'];
```
Comment (lpatiny) [inline src/encodeExperiment.js:13]:
```suggestion
      let min = array[i].min;
```

File: src/encodeExperiment.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      let max = array[i]['max'];
+      let nbSamples = array[i]['nbSamples'];
+      let sampling = array[i]['sampling'];
```
Comment (lpatiny) [inline src/encodeExperiment.js:16]:
```suggestion
      let sampling = array[i].sampling;
```

Same should be done everywhere

File: src/encodeExperiment.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        experiment[dict[index]['label']] =
+          dict[index]['values'][arrayOfExperiments[i][j]];
+        index++;
+      }
+      returnArrayOfExperiments.push(experiment);
+    }
+    return returnArrayOfExperiments;
+  }
+
+  return { returnArray, decodeExperiment };
```
Comment (lpatiny) [inline src/encodeExperiment.js:57]: We could rename the variable:
- returnArray => experiments

**Review by lpatiny — CHANGES_REQUESTED** [review]

**Review by kjappelbaum — APPROVED** [review]

**Pull request merged**